### DISCUSSION TITLE
Deploying flows that are built on the run time

### DISCUSSION THREAD

**Discussion by YasmineHamdy** [PrefectHQ/prefect#6752] [Q&A]
I have use case where I build the flows dynamically on the runtime with the tasks assigned to them depending on some configuration which the user define,

For example, 
```python
for flow_index, flow_tasks in flows.items():
  @flow(name=f"pipeline_{pipeline_id}_flow_{flow_index}", task_runner=ConcurrentTaskRunner())
  def flow_main():
    for task_config in flow_tasks:
      task = create_task(task_config['type'], name=f"task_{task_config['id']}")
      task.submit(data=flow[task_config['id']], return_state=True)
```

How would i deploy the those flows? 
Is there a way to deploy the flow using the name given in the args?

**Comment by jmg-duarte** [comment]
Is this for Prefect 2?
